views/includes/page-templates/beneficial-owner-nature-of-control-types.html

```diff
@@ -9,9 +9,16 @@
 {% set errorMessage = errors.beneficial_owner_nature_of_control_types if errors %}
 {% include "includes/inputs/nature-of-control-checkboxes/beneficial-owner-nature-of-control-types.html" %}
 
-{% if not isBeneficialOwnerGov and not entity_number %}
-  {% set errorMessage = {} if errors.beneficial_owner_nature_of_control_types %}
-  {% include "includes/inputs/nature-of-control-checkboxes/trustees-nature-of-control-types.html" %}
+{% if entity_number %}
```

Comment:
Tell me if I'm wrong.. but we want to see Trustee NOC if:
- they're not a BO gov 
AND
- in registration OR the flag is enabled?

So could this block/condition be simplified into...?

```
{% if not isBeneficialOwnerGov and (pageParams.isRegistration or pageParams.isUpdateTrustsFeatureEnabled) %}
  {% set errorMessage = {} if errors.beneficial_owner_nature_of_control_types %}
  {% include "includes/inputs/nature-of-control-checkboxes/trustees-nature-of-control-types.html" %}
{% endif %}
```

Also a couple of small other comments on naming in that suggestion:
- I'd recommend instead of just using the existence of the entity_number in the condition to signify being in registration or update, abstract that 'logic' away in a variable, i.e. `isRegistration` for example, to make reading the condition at a glance a little easier
- There is already an `isTrustFeatureEnabled` parameter in other pages/controllers using the feature flag introduced months ago. Even though it's not a direct conflict, distinguishing between the two by mentioning `Update` in the name could again make it clearer what flag is referenced here

Reply:
@conuirchgov there is a known "quirk" where if we submit a page in update journey and it triggers validation, when the page re-renders - you won't have access to the page params you passed in on the first load. This makes it tricky to use `isRegistration` flags in all the views. Happy to go into more detail if you want - but @cdrainCH let's use entity_number for now.

Reply:
Right... ok. That's interesting - never thought of that but I suppose that makes sense given the validation setup. 😬 So it's only `appData` we keep access to?

Tbh I didn't even mean to use pageParams, I'd just seen that as an example elsewhere, but even as a 

`{% set isRegistration = !entity_number %}`

in this template.

Reply:
Dismiss both naming points then 😂

Still think it would be worth simplifying the condition down to just to one expression.